app/views/flood_risk_engine/enrollments/addresses/edit.html.erb

```diff
@@ -0,0 +1,29 @@
+<%= link_to(
```

Comment:
OK - I see where you are coming from. 

The problem as I see it, is that the bits that are different between the two, aren't in easy places to strip out (within the form and within grid row div)

I think it's also a level of abstraction thing again. Having partials within partials can be hard to follow.

However, the back button code is the same between both views and the step show view - that needs to be extracted. Especially as all three have the same error - they use "Back" directly rather than pulling it in from a locale file!